ex3/k_namizaki/main3.py

```diff
@@ -0,0 +1,191 @@
+"""
+This code is to do regression analyzed using the least squares method.
+
+Calculate weights.
+Plot data.
+"""
+
+import argparse
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+def parse_args():
+    """
+    Get Arguments.
+
+    Returns
+    -------
+    parser.parse_args() : 引数を返す
+    """
+    parser = argparse.ArgumentParser(description="最小二乗法を用いて回帰分析を行う。")
+    parser.add_argument(
+        "-file",
+        help="ファイルを入力",
+        default=r"C:\Users\kyskn\B4Lecture-2024\ex3\k_namizaki\data2.csv",
+        type=str,
+    )
+    parser.add_argument("-n", help="次数", default=1, type=int)
+    parser.add_argument("-normal", help="正則化係数", default=0, type=int)
+    return parser.parse_args()
+
+
+def plot2d(x, y, w):
```

Comment:
関数の仮引数には，typing（型ヒント）を付けておくことをオススメします．[参考](https://docs.python.org/ja/3/library/typing.html)
これを付けておくと，VSCodeのsuggestionが働いてくれて，コーディングも捗ると思います．

```suggestion
from numpy.typing import ArrayLike

def plot2d(x: ArrayLike, y: ArrayLike, w: ArrayLike):
```

また，返り値を持つ場合や，デフォルト値を持つ関数の場合は以下のようになります．
```python
def hoge(arg1: int, arg2: str = "This Is Hoge") -> list[int]:
```